### ISSUE TITLE
Model Plugin - co2js

### ISSUE THREAD

**Issue by jawache** [Green-Software-Foundation/if#156]
- **What**: Create a model to interact with the CO2.js library
- **Why**: So we can use this model for network transfer estimation and other use cases where co2.js is used.
- **DoD**:
- [ ] Does the model meet the specification?
- [ ] Does the model have complete test coverage, including edge cases, happy path and error cases?
- [ ] Has documentation been written for someone writing an IMPL using this model?
- [ ] Has the `units.yml` been updated to take into consideration the params used in this model

## Config
- `model`: swd | 1byte as per the documentation https://developers.thegreenwebfoundation.org/co2js/models/

## Observation
_List the variables this model expects in an input observation_
- `bytes-transmitted`: The number of bytes transmitted over a network
- `green-web-host`: **Optional** Boolean, is the host the data is coming from defined as a green web host according to the green web foundation?

## Impacts
_List the variables this model exports out as an impact_
- `carbon`: Amount of carbon in gCO2e emitted for this network transfer according to the CO2js model.

## Behavior
_What does the model do to turn the observations into impacts? List any important equations or references._

- See: https://developers.thegreenwebfoundation.org/co2js
- We're leveraging CO2.js model to use the network carbon calculations. 
- Converting bytes transmitted to carbon.

## Assumptions
_What are the core assumptions this model makes?_
- There are many assumptions and the methodology is documented here: https://developers.thegreenwebfoundation.org/co2js/explainer/methodologies-for-calculating-website-carbon/

## Limitations
In what situations is this model limited in it's function, behaviour, accuracy etc...
- N/A

## Example
Provide at least one example of some input config and observations. This is in pimpl format not rimpl format, so the config is the global config and all parameters must be present on the observation.

### Input

```yaml
```yaml
config:
   model: swd
observations: 
  - timestamp: 2023-07-06T00:00
    duration: 3600
    bytes-transmitted: 100000
```

### Output

```yaml
config:
   model: swd
observations: 
  - timestamp: 2023-07-06T00:00
    duration: 3600
    bytes-transmitted: 100000
impacts: 
  - timestamp: 2023-07-06T00:00
    duration: 3600
    bytes-transmitted: 100000
    carbon: 10
```

**Comment by jmcook1186** [comment]
backlogged until after alpha release

**Comment by jmcook1186** [comment]
expected 4-5 hours of work remaining

**Comment by gnanakeethan** [comment]
@jmcook1186 I am finalizing the model spec and the goals. I had a thought about the following in which this model can utilise existing data instead of relying on the internal data from CO2.JS

Since the original issue was posted, our models have become complex. For example, we can now capture grid-intensity for a specific area using watt-time instead of relying on the internal data of CO2.JS. 

Just wondering whether should we address those concerns or just stick with the predetermined inputs & outputs defined in the original model specification above? 

cc: @jawache

**Comment by jmcook1186** [comment]
Nice idea. I would prefer to have options, especially since Watttime data is gated by API keys. 

Appreciating that Watttime data is more granular, I would still like users to be able to use the CO2js model plugin without a Watttime dependency.

Perhaps ship with CO2js data, and start a new ticket to integrate Watttime as an optional feature to it later?
